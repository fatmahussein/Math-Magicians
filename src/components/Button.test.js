import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Button from './button';

describe('Button Component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Button symbol="+" color="blue" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('calls onClick function when clicked', () => {
    const onClickMock = jest.fn();
    const { getByText } = render(<Button symbol="+" color="blue" onClick={onClickMock} />);
    const button = getByText('+');
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalled();
  });
});
