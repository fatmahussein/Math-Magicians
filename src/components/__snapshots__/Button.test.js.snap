// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="btn-wrapper"
    style="background-color: blue;"
    type="button"
  >
    +
  </button>
</DocumentFragment>
`;
